modules/lvm2/udiskslinuxlogicalvolume.c

```diff
@@ -380,12 +376,42 @@ handle_delete (UDisksLogicalVolume   *_volume,
     }
 
   /* Policy check. */
-  UDISKS_DAEMON_CHECK_AUTHORIZATION (daemon,
-                                     UDISKS_OBJECT (object),
+  UDISKS_DAEMON_CHECK_AUTHORIZATION (*daemon,
+                                     UDISKS_OBJECT (*object),
                                      lvm2_policy_action_id,
                                      options,
-                                     N_("Authentication is required to delete a logical volume"),
+                                     auth_err_msg,
                                      invocation);
+  rc = TRUE;
+ out:
+  return rc;
+}
+
+static gboolean
+handle_delete (UDisksLogicalVolume   *_volume,
+               GDBusMethodInvocation *invocation,
+               GVariant              *options)
+{
+  GError *error = NULL;
+  UDisksLinuxLogicalVolume *volume = UDISKS_LINUX_LOGICAL_VOLUME (_volume);
+  UDisksLinuxLogicalVolumeObject *object = NULL;
+  UDisksDaemon *daemon = NULL;
+  uid_t caller_uid;
+  gboolean teardown_flag = FALSE;
+  UDisksLinuxVolumeGroupObject *group_object;
+  LVJobData data;
+
+  g_variant_lookup (options, "tear-down", "b", &teardown_flag);
+
+  if (!common_setup (volume,
+                     invocation,
+                     options,
+                     N_("Authentication is required to delete a logical volume"),
+                     &object,
+                     &daemon,
+                     &caller_uid,
+                     NULL))
```

Comment:
I think the first three arguments could go on a single line and so could the ``out_`` ones. But YMMV, as always. :)